### PR TITLE
Deprecate DataProfileResult and DataQualityResult for Dataplex Datascans.

### PR DESCRIPTION
**Pull request by saurabh-net**
Deprecate DataProfileResult and DataQualityResult for Dataplex Datascans.
Part of https://github.com/hashicorp/terraform-provider-google/issues/15448

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [X] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/develop/run-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read [Write release notes](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:deprecation
dataplex: deprecated the following `google_dataplex_datascan` fields: `dataProfileResult` and `dataQualityResult`

```

### CODE REVIEW

**Review by hao-nan-li — COMMENTED** [review]
Looks like there's some conflicts to be resolved. Was the decision about not removing the fields from the yaml, but marked them with a deprecated flag?

**Comment by saurabh-net** [comment]
We are planning to remove the fields, but we need to deprecate them first per the discussion here: https://github.com/GoogleCloudPlatform/magic-modules/pull/8583#issuecomment-1671629701

**Comment by hao-nan-li** [comment]
> We are planning to remove the fields, but we need to deprecate them first per the discussion here: [#8583 (comment)](https://github.com/GoogleCloudPlatform/magic-modules/pull/8583#issuecomment-1671629701)

Sounds good, adding @c2thorn as reviewer here for visibility.

**Comment by saurabh-net** [comment]
Let me update this to resolve the conflicts in a bit.

File: mmv1/third_party/terraform/website/docs/guides/version_5_upgrade.html.markdown [hunk 1]
```diff
@@ -131,3 +131,9 @@ These two unsupported fields were introduced incorrectly. They are now removed.
 ### `liveness_probe.tcp_socket` is now removed
 
 This unsupported field was introduced incorrectly. It is now removed.
+
+## Resource: `google_dataplex_datascan`
+
+### `dataQualityResult` and `dataProfileResult` output fields are now removed 
+
+`dataQualityResult` and `dataProfileResult` were output-only fields which listed results for the latest job created under a Datascan. This is unlikely to be relevant in a Terraform context.
```
[the remark this answers is not in the document]
Comment (saurabh-net) [reply mmv1/third_party/terraform/website/docs/guides/version_5_upgrade.html.markdown]: Done

**Review by c2thorn — APPROVED** [review]

**Pull request merged**